Tests/MongoSwiftSyncTests/UnifiedTestRunner/EntityDescription.swift

```diff
@@ -115,3 +117,155 @@ enum EntityDescription: Decodable {
         }
     }
 }
+
+/// Wrapper around a MongoClient used for test runner purposes.
+struct UnifiedTestClient {
+    let client: MongoClient
+
+    fileprivate let commandMonitor: UnifiedTestCommandMonitor
+
+    init(_ clientDescription: EntityDescription.Client) throws {
+        let connStr = MongoSwiftTestCase.getConnectionString(
+            singleMongos: clientDescription.useMultipleMongoses != true
+        ).toString()
+        var opts = clientDescription.uriOptions ?? MongoClientOptions()
+        // If the test might execute a configureFailPoint command, for each target client the test runner MAY
+        // specify a reduced value for heartbeatFrequencyMS (and minHeartbeatFrequencyMS if possible) to speed
+        // up SDAM recovery time and server selection after a failure; however, test runners MUST NOT do so for
+        // any client that specifies heartbeatFrequencyMS in its uriOptions.
+        if opts.heartbeatFrequencyMS == nil {
+            opts.minHeartbeatFrequencyMS = 50
+            opts.heartbeatFrequencyMS = 50
+        }
+        self.client = try MongoClient.makeTestClient(connStr, options: clientDescription.uriOptions)
```

Comment:
we should use `opts` here instead of going back to the uriOptions one in case we set heartbeat frequency stuff.

Reply:
Yeah, I made the same mistake back during the libmongoc upgrade, and it did cause me a lot of confusion lol.